numba/core/analysis.py

```diff
@@ -414,6 +430,21 @@ def resolve_input_arg_const(input_arg_idx):
                 if(prune_stat):
                     # add the condition to the list of nullified conditions
                     nullified_conditions.append((condition, taken))
+        else:
+            # see if this is a branch on a constant value predicate
+            resolved_const = Unknown()
+            try:
+                resolved_const = find_const(func_ir, branch.cond)
+                if resolved_const is None:
+                    resolved_const = types.NoneType('none')
+            except GuardException:
```

Comment:
why not use `guard(find_const, ...)`?

Reply:
This is the same as the branch on l417, it's because `find_const` returns the actual python constant value and the constant value itself might be `None` and `guard(find_const, ...)` would alias "Return None because the constant is None" and "Return None because there was a GuardException". I think this is still valid.

Reply:
ah, ok